integration-libs/opf/base/core/services/opf-cart-handler.service.ts

```diff
@@ -51,23 +50,134 @@ export class OpfCartHandlerService {
   protected eventService = inject(EventService);
   protected checkoutBillingAddressFacade = inject(CheckoutBillingAddressFacade);
   protected opfGlobalMessageService = inject(OpfGlobalMessageService);
+  protected opfMiniCartComponentService = inject(OpfMiniCartComponentService);
+  protected initiateCartHandlerState: CartHandlerState = {
+    cartId: '',
+    userId: '',
+    previousCartId: '',
+  };
+  protected cartHandlerState: CartHandlerState = {
+    ...this.initiateCartHandlerState,
+  };
 
-  addProductToCart(
+  protected addProductToNewCart(
+    productCode: string,
+    quantity: number,
+    originalCartId: string,
+    pickupStore?: string | undefined
+  ): Observable<boolean> {
+    this.cartHandlerState.previousCartId = originalCartId;
+
+    return this.multiCartFacade
+      .createCart({
+        userId: this.cartHandlerState.userId,
+        extraData: { active: false },
+      })
+      .pipe(
+        take(1),
+        switchMap((cart: Cart) => {
+          if (!cart?.code) {
+            return throwError('CartId missing from new created cart');
+          }
+          this.cartHandlerState.cartId = cart.code;
+          this.multiCartFacade.addEntry(
+            this.cartHandlerState.userId,
+            this.cartHandlerState.cartId,
+            productCode,
+            quantity,
+            pickupStore
+          );
+          return this.checkStableCart(this.cartHandlerState.cartId);
+        }),
+        switchMap(() => {
+          this.multiCartFacade.loadCart({
+            cartId: this.cartHandlerState.cartId,
+            userId: this.cartHandlerState.userId,
+            extraData: { active: true },
+          });
+          return this.checkStableCart(this.cartHandlerState.cartId);
+        })
+      );
+  }
+
+  protected addProductToActiveCart(
     productCode: string,
     quantity: number,
     pickupStore?: string | undefined
   ): Observable<boolean> {
     this.activeCartFacade.addEntry(productCode, quantity, pickupStore);
-    return this.checkStableCart();
+    return this.checkStableCart().pipe(
+      switchMap(() => this.getCurrentCartId()),
+      map((cartId) => {
+        this.cartHandlerState.cartId = cartId;
+        return true;
+      })
+    );
+  }
+
+  allowMiniCartUpdate() {
```

Comment:
Would you consider creating a mirror function that is more reusable within the cart-handler service?

```
blockMiniCartComponentUpdate(decision: boolean) {
    this.opfMiniCartComponentService.blockUpdate(decision);
}
```